### ISSUE TITLE
Sorting by popular count will sort by ascending popularity

### ISSUE THREAD

**Issue by gannebamm** [GeoNode/geonode#12788]
## Expected Behavior
Sorting by popular count will sort items by **descending** popularity. E.g. the most popular item is the first to be shown.

## Actual Behavior
Sorting by popular count will sort items by _ascending_ popularity. Therefore the least popular item is the first to be shown. This is counter-intuitive.

## Steps to Reproduce the Problem

  1. Check popularity in Django admin (formerly it was displayed in Frontend)
  2. Use catalogue sorting to sort by popularity
  3. See popular ressources being listed last

or by API

- https://stable.demo.geonode.org/api/v2/resources?page=1&sort%5B%5D=popular_count

popular count is '0'

- https://stable.demo.geonode.org/api/v2/resources?page=703&sort%5B%5D=popular_count

popular count is > 7000

## Specifications

  - GeoNode version: 4.4.2 (stable)
  - Installation type (vanilla, geonode-project): 
  - Installation method (manual, docker):
  - Platform:
  - Additional details:

**Comment by giohappy** [comment]
@gannebamm you're right, the sorting should be descending and the fix is quick. It's just a matter of adding a minus sign to the sort parameter (h`ttps://stable.demo.geonode.org/api/v2/resources?page=1&sort[]=-popular_count`).

The point is, can we rely on the `popular_count` attribute? I have investigated a bit and it doesn't seem to work anymore. The count is incremented by legacy code paths that are not hit anymore (at least for the most part if not totally).

Counting views is more of an analytics feature. We have been discussing to implement again analytics support inside GeoNode, although backed by external services. I have been experimenting with Plausible and some vanilla JS frontend code connected to the MapStore events API. It works, but it needs funding to be fully implemented and published.

I would be inclined to remove `popular_count` from GeoNode 5, unless someone wants resurrect it.

**Comment by giohappy** [comment]
Closing as part of a repository cleanup: this issue was last updated before 2025-01-01. Not a judgment on the report itself — feel free to reopen or file a fresh issue against `main` if it's still relevant. Thanks for the report.
